addons/website_sale/models/website.py

```diff
@@ -792,15 +792,20 @@ def _get_checkout_steps(self):
         return steps
 
     def _get_checkout_step_values(self, href=None):
-        href = href or request.httprequest.path
+        def rewrite(path):
+            return self.env['ir.http'].url_rewrite(path)[0]
+        href = rewrite(href) if href else request.httprequest.path
         # /shop/address is associated with the delivery step
-        if href == '/shop/address':
+        if href == rewrite('/shop/address'):
             href = '/shop/checkout'
```

Comment:
Hi :) Sorry if I didn't understand the method, but shouldn't we also set the href to 'rewrite('shop/checkout')' to match line 805?

Reply:
yes, you are correct I missed it because I was thinking only about fixing the "current_website_checkout_step_href" (that was causing the 500 error) but with that correction we have the correct next/previous variables 👍 

I applied that correction.  I also checked to add a test for that, but it seems that these variable (previous*, next*) are only used in the `website_sale.navigation_buttons` template that is only used on /shop/cart template.

Reply:
I've added steps that test in the rendered page the previous_website_checkout_step and next_website_checkout_step variable in the case of /shop/address.